Makefile

```diff
@@ -473,6 +473,7 @@ test : fmt $(.DEFAULT_GOAL)
 		export GIT_CONFIG_NOSYSTEM=1; \
 		$(GO) test -count=1 $(GO_TEST_EXTRA_ARGS) $(addprefix ./,$(PKGS)); \
 		RET=$$?; \
+		chmod -R u+w "$$tempdir"; \
```

Comment:
This seems like most limited solution we can apply.  I've seen security problems from loosening permissions before deleting, but I don't think we have that here because it only applies to the user and not anyone else.

Do we know how this works on Windows?

Reply:
Hmm, Windows ... good question.  The Windows CI jobs seem to have passed, and I see these commands in the log output ahead of successful Go tests (e.g., in the ` Run GOPATH="$HOME/go" PATH="$HOME/go/bin:$PATH" script/cibuild` section of the `Build on Windows` job [output](https://github.com/git-lfs/git-lfs/pull/4157/checks?check_run_id=766167307)).  That's running under Mingw-w64, I think?

Reply:
If the answer is, “It doesn't break the testsuite and is no worse off than before,” which it looks like it is, then I'm fine with that.

Reply:
Cool.  And at least on some CI jobs, I see this is failing as I experienced it locally; see this recent job [output](https://github.com/git-lfs/git-lfs/runs/766491205), for instance.  So I think it'll be helpful to merge this, which I'll do in a moment.

Reply:
I'm very curious as to why it's failing now and wasn't before.  I guess that's the question I always ask about our CI jobs, though, so :man_shrugging:.

Reply:
Yeah, dunno.  I suppose the change could be upstream somewhere in the Go ecosystem, since we install a bunch of Go modules into this temporary directory, and the troublesome permissions are on certain directories in that hierarchy, e.g., `.../go/pkg/mod/gopkg.in/yaml.v2@v2.2.8` and so forth.